app/Services/TypeSenseScoutSearchService.php

```diff
@@ -65,6 +65,11 @@ public function search(string $q, ?string $orderByField = null,
                     $options['sort_by'] = "_text_match:desc,$orderByField:" . ($sortDirectionDescending ? "desc" : "asc");
                 }
 
+                // if order_by is a user supplied value make it a priority over _text_match
+                if ($orderByField !== 'members' && !is_null($orderByField)) {
```

Comment:
this `$oderByField !== 'members'` doesn't belong here, this should belong to another class/trait which exposes the method to apply things like this.

Reply:
@pushrbx could you show me how this would be done?

Reply:
This entirely belongs to the model I think. So the anime model's `getSearchIndexSortBy()` function's signature should probably change from
```php
public function getSearchIndexSortBy(): array|null
```
to
```php
public function getSearchIndexSortBy($orderByField, $sortDirectionDescending): array|null
```
and implement this logic accordingly. If the logic is same for both anime and manga, the logic should live in a service class i guess, and resolve the instance of that class inside the model. I can change this unless you want to take this on yourself. 😄

Reply:
@pushrbx the logic would be same for anime and manga, yep. So would need that service class.

If you could make this change, that would be very helpful 🙏

Reply:
Will do 👍

Reply:
I'm going to resolve this in a more simpler way.
By default when the user searches we want to show hits by relevancy, and then order them by "members".
When the `order_by` parameter is present, we don't want to show hits by relevancy anymore, the hits should be ordered by the field in the `order_by` parameter, and the relevancy score should be used for tie breaking.

So this bit:

```php
// override ordering field
if (!is_null($orderByField)) {
    $options['sort_by'] = "_text_match:desc,$orderByField:" . ($sortDirectionDescending ? "desc" : "asc");
}

// if order_by is a user supplied value make it a priority over _text_match
if ($orderByField !== 'members' && !is_null($orderByField)) {
    $options['sort_by'] = "$orderByField:" . ($sortDirectionDescending ? "desc" : "asc") . ",_text_match:desc";
}
```
Changes to this:
```php
// override ordering field
if (!is_null($orderByField)) {
    $options['sort_by'] = "$orderByField:" . ($sortDirectionDescending ? "desc" : "asc") . ",_text_match:desc";
}
```